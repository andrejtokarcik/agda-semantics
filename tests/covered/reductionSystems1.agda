module reductionSystems1 where


data ℕ : Set where 
  Z : ℕ  
  S : ℕ -> ℕ

_+_ : ℕ -> ℕ -> ℕ
n + Z   = n
n + S m = S (n + m)

_*_ : ℕ -> ℕ -> ℕ
n * Z = Z
n * S m = (n * m) + n


{-
{-# BUILTIN NATURAL  ℕ  #-}
{-# BUILTIN ZERO     Z #-}
{-# BUILTIN SUC      S  #-}
{-# BUILTIN NATPLUS  _+_  #-}
{-# BUILTIN NATTIMES _*_  #-}
-}
