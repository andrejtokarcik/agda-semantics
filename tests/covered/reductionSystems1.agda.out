<k>
    .K
</k>
<mgu>
    subst(.KList)
</mgu>
<ctx-stack>
    .List
</ctx-stack>
<sig>
    Fun ( "_*_" ) |-> ( #symVariable(47) : Data ( "8469" ) ) -> ( 
      #symVariable(48) : Data ( "8469" ) ) -> Data ( "8469" )
    Fun ( "_+_" ) |-> ( #symVariable(4) : Data ( "8469" ) ) -> ( #symVariable(
      5) : Data ( "8469" ) ) -> Data ( "8469" )
    Data ( "8469" ) |-> Set ( 0 )
    Con ( "S" ) |-> ( #symVariable(3) : Data ( "8469" ) ) -> Data ( "8469" )
    Con ( "Z" ) |-> Data ( "8469" )
</sig>
<ctx>
    .Map
</ctx>
<meta>
    .Map
</meta>
<ncat>
    Name ( "8469" ) |-> Data ( "8469" )
    Name ( "S" ) |-> Con ( "S" )
    Name ( "Z" ) |-> Con ( "Z" )
    Name ( "_*_" ) |-> Fun ( "_*_" )
    Name ( "_+_" ) |-> Fun ( "_+_" )
</ncat>
<fun>
    Fun ( "_*_" ) |-> 'fc(((#symFunClauses(60) Var ( "n" )) Con ( "Z" )) =
       Con ( "Z" ),, ((#symFunClauses(93) Var ( "n" )) (Con ( "S" ) Var ( "m" 
      ))) = ((Fun ( "_+_" ) ((Fun ( "_*_" ) Var ( "n" )) Var ( "m" ))) Var ( 
      "n" )))
    Fun ( "_+_" ) |-> 'fc(((#symFunClauses(17) Var ( "n" )) Con ( "Z" )) = Var
       ( "n" ),, ((#symFunClauses(46) Var ( "n" )) (Con ( "S" ) Var ( "m" ))) 
      = (Con ( "S" ) ((Fun ( "_+_" ) Var ( "n" )) Var ( "m" ))))
</fun>
